torch_geometric/nn/aggr/attention.py

```diff
@@ -53,28 +54,21 @@ def reset_parameters(self):
         reset(self.gate_nn)
         reset(self.nn)
 
-    def forward(self, x: Tensor, batch: Optional[Tensor] = None,
-                size: Optional[int] = None) -> Tensor:
-        r"""
-        Args:
-            x (Tensor): The input node features.
-            batch (LongTensor, optional): A vector that maps each node to its
-                respective graph identifier. (default: :obj:`None`)
-            size (int, optional): The number of graphs in the batch.
-                (default: :obj:`None`)
-        """
-        if batch is None:
-            batch = x.new_zeros(x.size(0), dtype=torch.int64)
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+
+        self.assert_index_present(index)
```

Comment:
I don't think this is necessarily true. We can replace `scatter_add` with `self.reduce`, and the `softmax` call should support both `index` and `ptr`.

Reply:
you're completely correct, I missed it. Thanks, updated.